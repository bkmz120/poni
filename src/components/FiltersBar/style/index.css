.filtersBar {
  padding: 10px 0;
}

.filters {
  display: inline-block;
  vertical-align: top;
}

.filter {
  display: inline-block;
  vertical-align: top;
  width: 250px;
  margin-right: 20px;
}

.applyBtn {
  display: inline-block;
  vertical-align: top;
  line-height: 38px;
  color: #fff;
  padding: 0 20px;
  background: #162671;
  cursor: pointer;

  &:hover {
    opacity: 0.9;
  }
}
