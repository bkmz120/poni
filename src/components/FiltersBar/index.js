import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'
import { Tooltip } from 'react-tippy'
import 'react-tippy/dist/tippy.css'

import style from './style/index.css'

export default class FiltersBar extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      minMaxError: false,
    }
  }

  handleChangeColor = current => {
    const {
      changeValues,
      values
    } = this.props

    let color = ''
    if (current) {
      color = current.value
    }
    changeValues({
      ...values,
      color,
    })
  }

  handleChangeKind = current => {
    const {
      changeValues,
      values
    } = this.props

    let kind = ''
    if (current) {
      kind = current.value
    }
    changeValues({
      ...values,
      kind,
    })
  }

  handleChangeMinPrice = e => {
    const value = e.target.value
    const {
      changeValues,
      values,
    } = this.props
    this.setState({
      minMaxError: false,
    })
    if (/^\d*\.*\d*$/.test(value)) {
      changeValues({
        ...values,
        minPrice: value,
      })
    }
  }

  handleChangeMaxPrice = e => {
    const value = e.target.value
    const {
      changeValues,
      values,
    } = this.props
    this.setState({
      minMaxError: false,
    })
    if (/^\d*\.*\d*$/.test(value)) {
      changeValues({
        ...values,
        maxPrice: value,
      })
    }
  }

  handleChangeIsNew = (e) => {
    const value = e.target.checked
    const {
      changeValues,
      values,
    } = this.props
    changeValues({
      ...values,
      isNew: value,
    })
  }

  handleClickApply = () => {
    const {
      values,
      apply
    } = this.props
    if (values.minPrice !== '' &&
        values.maxPrice !== '' &&
        parseFloat(values.minPrice) >= parseFloat(values.maxPrice)
    ) {
      this.setState({
        minMaxError: true,
      })
    } else {
      apply()
    }
  }

  render () {
    const {
      colorsOptions,
      kindsOptions,
      values
    } = this.props

    let color
    if (values.color !== '') {
      color = {
        label: values.color,
        value: values.color,
      }
    }

    let kind
    if (values.kind !== '') {
      kind = {
        label: values.kind,
        value: values.kind,
      }
    }

    const {
      minPrice,
      maxPrice,
      isNew,
    } = values

    return <div className={style.filtersBar}>
      <div className={style.filters}>
        <div className={style.filter}>
          <Select
            isClearable={true}
            onChange={this.handleChangeColor}
            options={colorsOptions}
            placeholder="Цвет"
            value={color}
          />
        </div>
        <div className={style.filter}>
          <Select
            isClearable={true}
            onChange={this.handleChangeKind}
            options={kindsOptions}
            placeholder="Вид"
            value={kind}
          />
        </div>
        <div className={style.filter}>
          <span className={style.priceLabel}>Цена от</span>
          <Tooltip
            title="Минимальное значение должно быть меньше максимального"
            position="bottom"
            open={this.state.minMaxError}
            arrow={true}
            trigger="manual"
          >
            <input
              className={style.priceValue}
              type='text'
              value={minPrice}
              onChange={this.handleChangeMinPrice}
            />
          </Tooltip>
          <span className={style.priceLabel}>до</span>
          <input
            className={style.priceValue}
            type='text'
            value={maxPrice}
            onChange={this.handleChangeMaxPrice}
          />
        </div>
      </div>
      <div className={style.filter}>
        <label className={style.checkbox}>
          <input
            className={style.checkboxValue}
            type="checkbox"
            checked={isNew}
            onChange={this.handleChangeIsNew}
          />
          только новинки
        </label>
      </div>
      <div
        className={style.applyBtn}
        onClick={this.handleClickApply}
      >
        Найти
      </div>
    </div>
  }
}

FiltersBar.propTypes = {
  colorsOptions: PropTypes.array.isRequired,
  kindsOptions: PropTypes.array.isRequired,
  values: PropTypes.shape({
    color: PropTypes.string,
    kind: PropTypes.string,
    minPrice: PropTypes.string,
    maxPrice: PropTypes.string,
    isNew: PropTypes.bool,
  }).isRequired,
  changeValues: PropTypes.func.isRequired,
  apply: PropTypes.func.isRequired,
}
