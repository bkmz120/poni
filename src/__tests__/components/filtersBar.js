/* eslint-disable */

import FiltersBar from '@components/FiltersBar'

jest.mock('react-tippy/dist/tippy.css', () => (''))

describe('FiltersBar',()=>{
  const colorsOptions = [
    {
      label:'Красный',
      value:'Красный',
    },
    {
      label:'Синий',
      value:'Синий',
    },
    {
      label:'Фиолетовый',
      value:'Фиолетовый',
    },
    {
      label:'Черный',
      value:'Черный',
    },
  ]

  const kindsOptions = [
    {
      label:'Аликорн',
      value:'Аликорн',
    },
    {
      label:'Единорог',
      value:'Единорог',
    },
    {
      label:'Земная пони',
      value:'Земная пони',
    },
    {
      label:'Пегас',
      value:'Пегас',
    },
  ]

  it('check init snapshot', () => {
    const values = {
      color: '',
      kind: '',
      minPrice: '',
      maxPrice: '',
      isNew: false,
    }

    const changeValues = () => {}
    const apply = () => {}

    const FiltersBarWrapper = shallow(
        <FiltersBar
          colorsOptions={colorsOptions}
          kindsOptions={kindsOptions}
          values={values}
          changeValues={changeValues}
          apply={apply}
        />
    );
    expect(FiltersBarWrapper).toMatchSnapshot()
  })

  it('should fire apply on apply btn click', () => {
    const values = {
      color: '',
      kind: '',
      minPrice: '',
      maxPrice: '',
      isNew: false,
    }

    const changeValues = () => {}
    const apply = jest.fn()

    const FiltersBarWrapper = shallow(
        <FiltersBar
          colorsOptions={colorsOptions}
          kindsOptions={kindsOptions}
          values={values}
          changeValues={changeValues}
          apply={apply}
        />
    );
    FiltersBarWrapper.find('.applyBtn').simulate('click')
    expect(apply).toHaveBeenCalled()
  })
})

