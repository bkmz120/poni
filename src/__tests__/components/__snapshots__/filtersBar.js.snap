// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersBar check init snapshot 1`] = `
<div
  className="filtersBar"
>
  <div
    className="filters"
  >
    <div
      className="filter"
    >
      <StateManager
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isClearable={true}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Красный",
              "value": "Красный",
            },
            Object {
              "label": "Синий",
              "value": "Синий",
            },
            Object {
              "label": "Фиолетовый",
              "value": "Фиолетовый",
            },
            Object {
              "label": "Черный",
              "value": "Черный",
            },
          ]
        }
        placeholder="Цвет"
      />
    </div>
    <div
      className="filter"
    >
      <StateManager
        defaultInputValue=""
        defaultMenuIsOpen={false}
        defaultValue={null}
        isClearable={true}
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Аликорн",
              "value": "Аликорн",
            },
            Object {
              "label": "Единорог",
              "value": "Единорог",
            },
            Object {
              "label": "Земная пони",
              "value": "Земная пони",
            },
            Object {
              "label": "Пегас",
              "value": "Пегас",
            },
          ]
        }
        placeholder="Вид"
      />
    </div>
    <div
      className="filter"
    >
      <span
        className="priceLabel"
      >
        Цена от
      </span>
      <Tooltip
        animateFill={true}
        animation="shift"
        arrow={true}
        arrowSize="regular"
        className=""
        delay={0}
        disabled={false}
        distance={10}
        duration={375}
        followCursor={false}
        hideDelay={0}
        hideDuration={375}
        hideOnClick={true}
        html={null}
        inertia={false}
        interactive={false}
        interactiveBorder={2}
        multiple={false}
        offset={0}
        onHidden={[Function]}
        onHide={[Function]}
        onRequestClose={[Function]}
        onShow={[Function]}
        onShown={[Function]}
        open={false}
        popperOptions={Object {}}
        position="bottom"
        size="regular"
        sticky={false}
        stickyDuration={200}
        style={Object {}}
        theme="dark"
        title="Минимальное значение должно быть меньше максимального"
        touchHold={false}
        trigger="manual"
        unmountHTMLWhenHide={false}
      >
        <input
          className="priceValue"
          onChange={[Function]}
          type="text"
          value=""
        />
      </Tooltip>
      <span
        className="priceLabel"
      >
        до
      </span>
      <input
        className="priceValue"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="filter"
  >
    <label
      className="checkbox"
    >
      <input
        checked={false}
        className="checkboxValue"
        onChange={[Function]}
        type="checkbox"
      />
      только новинки
    </label>
  </div>
  <div
    className="applyBtn"
    onClick={[Function]}
  >
    Найти
  </div>
</div>
`;
